src/common/meta/src/rpc/procedure.rs

```diff
@@ -81,6 +81,8 @@ pub fn procedure_state_to_pb_state(state: &ProcedureState) -> (PbProcedureStatus
         ProcedureState::RollingBack { error } => {
             (PbProcedureStatus::RollingBack, error.to_string())
         }
+        // TODO(weny): define a new status for poisoned.
+        ProcedureState::Poisoned { error, .. } => (PbProcedureStatus::Failed, error.to_string()),
```

Comment:
Why not add it right now?

Reply:
Resolved in [3331fce](https://github.com/GreptimeTeam/greptimedb/pull/5822/commits/3331fce63408d6f5967551f116194a93890b094c)

Reply:
wait for https://github.com/GreptimeTeam/greptime-proto/pull/229